### ISSUE TITLE
Do you have any problems in running speed?

### ISSUE THREAD

**Issue by Ostnie** [Colhodm/TensorFlow-Classifier-Alex_Net#1]
Hello, I am a freshman in machine learning ,too. I used the  finetune_alexnet_with_tensorflow several days ago ,but I don't know how to get the unkonwn data's results , and I find your code and it help me a lot ,but now I have a problem about the program's running speed 。 With the same size of data, the forecast time is much longer than my training time ，Just as I konw ,I think the speed should be the same .So I don't konw why it come as this ,Do you have any advice?
I guess the reason is that when I trained,the batch size is 64 ,but when I predict, you know ,basic on your demo, I can only get one result each time ,I tried to change the Alexnet's code and I can get 64 results a time but I don't konw how to print the filename of the results. It's too annoying but I failed to get the filename. So do you know how to modify your code to input  several photo a time ?
Many thanks!

**Comment by Colhodm** [comment]
Hi Ostnie,
The prediction time should be much faster than your training time, since you're merely extracting weights and running a softmax on them.  So for example I think fine tuning the last two layers with 1500 images on 100 epochs with a batch size of 80 takes me about 2.5 hours with a pretty decent GPU. However prediction on 200 images, takes about 2 minutes. To get the filename of the results, what I would do is just how I write to an output file with the predictions/AUC/accuracy, you can get the appropriate file path from the list with the corresponding index to your test image. I stored all the file paths in a list, to make it easy to write what files it was predicting wrong and all that, if you look through the code you'll see that. As for your final question about testing batches of images, I think that it would make sense if you have a HUGE testing set e.g like the actual imagenet competition, but anything you're probably running can work with testing single images. If it helps too, I can look through your code and see what potentially might be wrong!
Thanks,
Arjun

**Comment by Ostnie** [comment]
Hi , judging from your data(1500 images on 100 epochs,the training efficiency is ten times than predict)
1500 image,100 epochs means 150000 in total and it takes 150 mins, so  1000 images are been calculated each min , but only 200 images takes 2 mins ,so , it is obvious  that their familiarity is ten times different ,and  I think it is no problem ,because every time you predict ,you should  run the total model and get only one value .
I agree with your point that it makes no difference for me because my data is small ,the reason for me to try to modify the code for batch input is that I want it look better ,isn't it? I just think I should have a try ,em.....,but you know I  failed .
I have another question about your code ,because I have just compared the code of our own, I notice you  used this 
softmax=tf.nn.softmax(score)
and I use the score directly as my predict value ,it seems make no difference in my data ,could you tell me it's  function?
Many thanks!

**Comment by Colhodm** [comment]
Hi,
so defining softmax=tf.nn.softmax(score) is just the proper way of coding in tensor flow, as you want to initialize variables before you utilize them during your session. I think that using a batch input would make sense if you were utilizing a large input of images, but since you're not I dont think its worth the headache!

**Comment by Ostnie** [comment]
Hi，maybe it isn't a problems for you to change a batch input but it seems difficult for me, I don't know how to change this from one to batch 
![image](https://user-images.githubusercontent.com/31840090/38197802-a3b3dd20-36bc-11e8-8522-826ec3463c90.png)
every time I changed it and I will get :Shape must be rank 0 but is rank 1 for 'DecodeJpeg' (op: 'DecodeJpeg') with input shapes: [1].
could please tell me how can I change this code ?  Really thanks !
